backend/hitas/views/housing_company.py

```diff
@@ -42,6 +42,7 @@ class HousingCompanyFilterSet(HitasFilterSet):
     postal_code = HitasPostalCodeFilter(field_name="postal_code__value")
     property_manager = HitasCharFilter(field_name="property_manager__name", lookup_expr="icontains")
     developer = HitasCharFilter(field_name="developer__value", lookup_expr="icontains")
+    legacy_id = HitasCharFilter(lookup_expr="iexact")
```

Comment:
`legacy_id` is numeric so `iexact` does not really make sense. but maybe not the currect datatype either

Reply:
The data stored in there is numeric, but the field is a CharField, so it doesn't really matter if its `iexact` or `exact`.

Reply:
`iexact` will result of `UPPER(legacy_id) = ?` instead of `legacy_id = ?`. From current data size perspective it does not matter that much but it's slower: conversion and if the column is indexed it would need to be indexed with `UPPER`, otherwise it will not be used